GoogleChromeLabs/ps-analysis-tool
# Feature: Vertical menu items

## Description

This PR aims to match the UI of the application tab, where a sidebar lists all the frames and clicking on each frame will list the cookies set by that particular frame.

## Relevant Technical Choices
1. Once used lands on the landing page it will show a landing page holder (will be replaced once issue #66 is completed).
2. Clicking on the cookies will open a list of all the frames which we get using `chrome.webNavigation.getAllFrames` we use this function on initial load, when storage is changed and when something in the tab changes. There might be cases where 2 different frameIds have the same URL in that case we store the frameIds in an array.
3. On loading the component after clicking on the frame we calculate the cookies which are set by the selected frame, by checking if the current frameId is available in cookies `frameIdList`.

## Testing Instructions
1. Open the terminal in the project folder.
2. Run `npm run dev`.
3. Open the `CNN` website, open the DevTools panel and click on `Privacy Sandbox`.
4. You should notice that the header menu has moved to the side and is not visible at first glance. Clicking on the `Cookies` tab will show an accordion of frames.
5. Select any frame that is shown and on the right side there will be a list of cookies. (Note if there is no cookie set by the frame then the main content will be empty).
6. On selecting the `Cookies` heading the accordion closes and the landing page placeholder reappears.

## Screenshot/Screencast

https://github.com/GoogleChromeLabs/ps-analysis-tool/assets/59614577/9483a9d5-4c5e-44fb-ab6a-55deffd10bad



---



Fixes #71
